media_driver/linux/common/os/i915/mos_bufmgr.c

```diff
@@ -4934,3 +5073,106 @@ int mos_set_context_param_load_balance(struct mos_linux_context *ctx,
     free(balancer);
     return ret;
 }
+
+int mos_set_context_param_bond_master(struct mos_linux_context *ctx)
+{
+    int ret = 0;
+    struct {
+        struct i915_user_extension base;
+        __u16 engine_index;
+        __u16 num_siblings;
+        __u32 flags;
+        __u64 mbz64;
+        struct i915_engine_class_instance engines[1];
+    } __attribute__((packed)) balancer;
```

Comment:
You could also use the I915_DEFINE_CONTEXT_ENGINES_LOAD_BALANCE macro from i915_drm.h so you don't need to duplicate the structure definition:

I915_DEFINE_CONTEXT_ENGINES_LOAD_BALANCE(balancer, 1);

Reply:
I don't want to use VLA as some compiler doesn't support VLA. But redefining is not good. I will rewrite it with malloc/free, similar to the  mos_set_context_param_load_balance() function.